sdk/core/Azure.Core/tests/AzureSasCredentialSynchronousPolicyTests.cs

```diff
@@ -132,5 +131,46 @@ public async Task VerifyRetryAfterSasCredentialUpdateOperation()
                 Assert.AreEqual((int)HttpStatusCode.OK, response.Status);
             }
         }
+
+        [Test]
+        public async Task VerifyRetryAfterMultipleSasCredentialUpdate()
+        {
+            // Arrange
+            const string INITIAL_QUERY_URI = "?foo=bar";
+            const string FIRST_SIGNATURE_VALUE = "?sig=first_signature_value";
+            const string SECOND_SIGNATURE_VALUE = "?sig=second_signature_value";
+            const string THIRD_SIGNATURE_VALUE = "?sig=third_signature_value";
+            int callCount = 0;
+            var azureSasCredential = new AzureSasCredential(FIRST_SIGNATURE_VALUE);
+            var transport = new MockTransport((req) =>
+            {
+                if (callCount++ == 0)
+                {
+                    Assert.AreEqual("?foo=bar&sig=first_signature_value", req.Uri.Query);
+                    Thread.Sleep(100);
```

Comment:
I'd suggest moving this into the RetryPolicyTests and using the pattern there with the AsyncGates to have more control over when the retry occurs.

Reply:
Fixed the existing tests instead.